BHoMUpgrader62/Converter.cs

```diff
@@ -38,12 +38,95 @@ public class Converter : Base.Converter
         public Converter() : base()
         {
             PreviousVersion = "6.1";
+
+            ToNewObject.Add("BH.oM.Lighting.Elements.Luminaire", UpgradeLuminaire);
         }
 
         /***************************************************/
         /**** Private Methods                           ****/
         /***************************************************/
 
+        private static Dictionary<string, object> UpgradeLuminaire(Dictionary<string, object> oldVersion)
+        {
+            Dictionary<string, object> newVersion = new Dictionary<string, object>(oldVersion);
+            if (newVersion.ContainsKey("Direction"))
+            {
+                if (newVersion["Direction"] is null)
+                {
+                    newVersion["Orientation"] = null;
+                    newVersion.Remove("Direction");
+                    return newVersion;
+                }
+
+                Dictionary<string, object> xVec = new Dictionary<string, object>() { ["X"] = 1, ["Y"] = 0, ["Z"] = 0 };
+                Dictionary<string, object> yVec = new Dictionary<string, object>() { ["X"] = 0, ["Y"] = 1, ["Z"] = 0 };
+                Dictionary<string, object> zVec = new Dictionary<string, object>() { ["X"] = 0, ["Y"] = 0, ["Z"] = 1 };
+
+                Dictionary<string, object> basis = new Dictionary<string, object>() { ["X"] = xVec, ["Y"] = yVec, ["Z"] = zVec };
+
+                Dictionary<string, object> orientation = newVersion["Direction"] as Dictionary<string, object>;
+                double dirX = (double)orientation["X"];
+                double dirY = (double)orientation["Y"];
+                double dirZ = (double)orientation["Z"];
```

Comment:
Might be worth doing these in `try/catch` statements in case any of them can't cast/convert to a `double`

Reply:
added